File: sdk/typescript/examples/08-eq-config.ts
// 08 — Equalizer & replaygain configuration
//
// Reads the current settings, prints a summary, then applies a "warm + bass
// boost" 5-band EQ preset and enables album-mode ReplayGain.
//
//   bun run examples/08-eq-config.ts

import { ReplaygainType } from '../src/index.js';
import { createClient } from './_client.js';

const client = createClient();

const before = await client.settings.get();
console.log(`Before:`);
console.log(`  EQ enabled       : ${before.eqEnabled}`);
console.log(`  EQ precut        : ${before.eqPrecut}`);
console.log(`  Replaygain type  : ${before.replaygainSettings.type}`);
console.log(`  Replaygain noclip: ${before.replaygainSettings.noclip}`);

await client.settings.save({
  eqEnabled: true,
  eqPrecut: -3,
  eqBandSettings: [
    { cutoff: 60,    q: 7, gain:  4 },  // bass boost
    { cutoff: 200,   q: 7, gain:  1 },
    { cutoff: 800,   q: 7, gain:  0 },
    { cutoff: 4_000, q: 7, gain: -2 },  // tame harshness
    { cutoff: 12_000, q: 7, gain:  2 }, // air
  ],
  replaygainSettings: {
    noclip: true,
    type: ReplaygainType.Album,
    preamp: 0,
  },
});

const after = await client.settings.get();
console.log(`\nAfter:`);
console.log(`  EQ enabled       : ${after.eqEnabled}`);
console.log(`  EQ precut        : ${after.eqPrecut}`);
console.log(`  Replaygain type  : ${after.replaygainSettings.type}`);
console.log(`  Replaygain noclip: ${after.replaygainSettings.noclip}`);

File: sdk/elixir/examples/08_eq_config.exs
# 08 — EQ configuration
#
# Enable the equalizer with a 5-band bass-boost / presence-cut profile.
#
#   mix run examples/08_eq_config.exs

Code.require_file("_helper.exs", __DIR__)

client = Examples.Helper.client()

:ok =
  Rockbox.Settings.update(client,
    eq_enabled: true,
    eq_precut: -3,
    eq_band_settings: [
      %{cutoff: 60, q: 7, gain: 3},
      %{cutoff: 200, q: 7, gain: 0},
      %{cutoff: 800, q: 7, gain: 0},
      %{cutoff: 4000, q: 7, gain: -2},
      %{cutoff: 12_000, q: 7, gain: 1}
    ]
  )

{:ok, settings} = Rockbox.Settings.get(client)

IO.puts("EQ enabled: #{settings.eq_enabled}")
IO.puts("EQ precut:  #{settings.eq_precut}")

for band <- settings.eq_band_settings do
  IO.puts("  #{String.pad_leading(Integer.to_string(band.cutoff), 6)} Hz  q=#{band.q}  #{band.gain} dB")
end
